### PR TITLE
HBASE-25401 Add trace support for async call in rpc client

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-22120 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 51s |  HBASE-22120 passed  |
| +1 :green_heart: |  checkstyle  |   1m 59s |  HBASE-22120 passed  |
| +1 :green_heart: |  spotbugs  |  16m 11s |  HBASE-22120 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 27s |  the patch passed  |
| -0 :warning: |  checkstyle  |   1m 59s |  root: The patch generated 1 new + 36 unchanged - 15 fixed = 37 total (was 51)  |
| -0 :warning: |  whitespace  |   0m  0s |  The patch has 2 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  xml  |   0m  3s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  17m 29s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1 3.3.0.  |
| +1 :green_heart: |  hbaseprotoc  |   6m 16s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |  17m 34s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   1m  5s |  The patch does not generate ASF License warnings.  |
|  |   |  79m 55s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2790 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle cc hbaseprotoc prototool xml |
| uname | Linux 56ae726d35d6 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-22120 / 4258d2c9ee |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/1/artifact/yetus-general-check/output/diff-checkstyle-root.txt |
| whitespace | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/1/artifact/yetus-general-check/output/whitespace-eol.txt |
| Max. process+thread count | 137 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-common hbase-client hbase-server . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=3.1.12 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 17s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-22120 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 32s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 12s |  HBASE-22120 passed  |
| +1 :green_heart: |  compile  |   3m  2s |  HBASE-22120 passed  |
| +1 :green_heart: |  shadedjars  |   7m 24s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 26s |  HBASE-22120 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  5s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 51s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 51s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m 18s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 36s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 146m  9s |  root in the patch failed.  |
|  |   | 189m 14s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2790 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux e16ccedd3fbe 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-22120 / 4258d2c9ee |
| Default Java | AdoptOpenJDK-11.0.6+10 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/1/testReport/ |
| Max. process+thread count | 4113 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-common hbase-client hbase-server . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 27s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-22120 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 34s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 44s |  HBASE-22120 passed  |
| +1 :green_heart: |  compile  |   2m 25s |  HBASE-22120 passed  |
| +1 :green_heart: |  shadedjars  |   6m 59s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 36s |  HBASE-22120 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 24s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 22s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 22s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m  4s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 35s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 266m 14s |  root in the patch failed.  |
|  |   | 303m 48s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2790 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 585d437cef3a 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-22120 / 4258d2c9ee |
| Default Java | AdoptOpenJDK-1.8.0_232-b09 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/1/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/1/testReport/ |
| Max. process+thread count | 3716 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-common hbase-client hbase-server . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
PTAL @saintstack 

This patch shows how we pass Span across different threads, and also how to propagate it through RPC.

And also added a UT to confirm that the above things works properly.

Thanks.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 23s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-22120 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 34s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 28s |  HBASE-22120 passed  |
| +1 :green_heart: |  checkstyle  |   1m 56s |  HBASE-22120 passed  |
| +1 :green_heart: |  spotbugs  |  16m  0s |  HBASE-22120 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 28s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 56s |  root: The patch generated 0 new + 32 unchanged - 6 fixed = 32 total (was 38)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  3s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  17m 10s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1 3.3.0.  |
| +1 :green_heart: |  hbaseprotoc  |   6m 19s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |  16m 50s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   1m  5s |  The patch does not generate ASF License warnings.  |
|  |   |  79m 13s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2790 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle cc hbaseprotoc prototool xml |
| uname | Linux 3ef7035eceb0 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-22120 / 20c59d8e29 |
| Max. process+thread count | 137 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-common hbase-client hbase-server . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/2/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=3.1.12 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-22120 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 30s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m  6s |  HBASE-22120 passed  |
| +1 :green_heart: |  compile  |   2m 53s |  HBASE-22120 passed  |
| +1 :green_heart: |  shadedjars  |   7m 28s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   5m 55s |  HBASE-22120 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 15s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 15s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   9m 43s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   5m 42s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 161m 21s |  root in the patch failed.  |
|  |   | 210m 21s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2790 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 8e2110d49255 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-22120 / 20c59d8e29 |
| Default Java | AdoptOpenJDK-11.0.6+10 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/2/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/2/testReport/ |
| Max. process+thread count | 4095 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-common hbase-client hbase-server . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-22120 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 32s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 44s |  HBASE-22120 passed  |
| +1 :green_heart: |  compile  |   2m 20s |  HBASE-22120 passed  |
| +1 :green_heart: |  shadedjars  |   6m 59s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 35s |  HBASE-22120 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 22s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 26s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 26s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m  5s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 34s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 336m 51s |  root in the patch failed.  |
|  |   | 374m 29s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2790 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 7396157276e1 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-22120 / 20c59d8e29 |
| Default Java | AdoptOpenJDK-1.8.0_232-b09 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/2/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/2/testReport/ |
| Max. process+thread count | 5535 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-common hbase-client hbase-server . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcServer.java [hunk 1]
```diff
@@ -182,21 +176,4 @@ public int getNumOpenConnections() {
     // allChannels also contains the server channel, so exclude that from the count.
     return channelsCount > 0 ? channelsCount - 1 : channelsCount;
   }
-
-  @Override
-  public Pair<Message, CellScanner> call(BlockingService service,
```
Comment (infraio) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcServer.java:187]: This is related to this issue?

Comment (Apache9) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcServer.java:187]: Oh, this should have already been landed on master. Let's check if we need a rebase.

**Review by infraio — APPROVED** [review]
+1

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-22120 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 40s |  HBASE-22120 passed  |
| +1 :green_heart: |  checkstyle  |   1m 54s |  HBASE-22120 passed  |
| +1 :green_heart: |  spotbugs  |  16m 38s |  HBASE-22120 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 43s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 31s |  root: The patch generated 0 new + 21 unchanged - 4 fixed = 21 total (was 25)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  3s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  21m 30s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1 3.3.0.  |
| +1 :green_heart: |  hbaseprotoc  |   7m 28s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |  20m 30s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   1m  4s |  The patch does not generate ASF License warnings.  |
|  |   |  91m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2790 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle cc hbaseprotoc prototool xml |
| uname | Linux 91320f389799 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-22120 / 6be338c6ca |
| Max. process+thread count | 137 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-common hbase-client hbase-server . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/3/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=3.1.12 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-22120 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 32s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 16s |  HBASE-22120 passed  |
| +1 :green_heart: |  compile  |   2m 54s |  HBASE-22120 passed  |
| +1 :green_heart: |  shadedjars  |   7m  3s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 27s |  HBASE-22120 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  6s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 48s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 48s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m 10s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 25s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 169m 39s |  root in the patch passed.  |
|  |   | 211m 54s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/3/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2790 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 4e51608d563d 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-22120 / 6be338c6ca |
| Default Java | AdoptOpenJDK-11.0.6+10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/3/testReport/ |
| Max. process+thread count | 6732 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-common hbase-client hbase-server . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 27s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-22120 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 32s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 39s |  HBASE-22120 passed  |
| +1 :green_heart: |  compile  |   2m 23s |  HBASE-22120 passed  |
| +1 :green_heart: |  shadedjars  |   6m 59s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 35s |  HBASE-22120 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 24s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 20s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 20s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 56s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 35s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 336m 42s |  root in the patch failed.  |
|  |   | 374m  7s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/3/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2790 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 9afe17c69351 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-22120 / 6be338c6ca |
| Default Java | AdoptOpenJDK-1.8.0_232-b09 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/3/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/3/testReport/ |
| Max. process+thread count | 5851 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-common hbase-client hbase-server . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2790/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**